### PR TITLE
chore: Change tests to circle CI

### PR DESCRIPTION
**Pull request by codedsun**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -108,6 +108,34 @@ jobs:
...
+                command: ./scripts/test_db.sh
+
+          - run:
+                command: pytest tests -v --cov=.
+
+          - run:
+                command: bash <(curl -s https://codecov.io/bash)
+                when: on_success
+          - run:
+                command: bash scripts/push_api_docs.sh
```
Comment (iamareebjamal) [inline .circleci/config.yml:136]: Use 2 spaces for indentation

File: .circleci/config.yml [hunk 2]
```diff
@@ -108,6 +108,34 @@ jobs:
...
+                PATH: $PATH:${HOME}/google-cloud-sdk/bin
+                DATABASE_URL: postgresql://test@localhost:5433/test
+                TEST_DATABASE_URL: postgresql://test@localhost:5433/test
+
+      working_directory: ~/code
+
+      steps:
+          - checkout
+
+          - run:
```
Comment (iamareebjamal) [inline .circleci/config.yml:126]: Restore cache step is missing

File: .circleci/config.yml [hunk 3]
```diff
@@ -108,6 +108,34 @@ jobs:
...
+                command: ./scripts/test_db.sh
```
Comment (iamareebjamal) [inline .circleci/config.yml:127]: DB and redis services are missing

File: .circleci/config.yml [hunk 4]
```diff
@@ -108,6 +108,52 @@ jobs:
...
 
+  test:
+      docker:
+          - image: circleci/python:3.7.4-node
+            environment:
+                APP_CONFIG: config.TestingConfig
+                SECRET_KEY: super secret key
+                PATH: $PATH:${HOME}/google-cloud-sdk/bin
+                DATABASE_URL: postgresql://test@localhost:5433/test
+                TEST_DATABASE_URL: postgresql://test@localhost:5433/test
```
Comment (iamareebjamal) [inline .circleci/config.yml:119]: Don't know where the env is coming from. Match dredd job

File: .circleci/config.yml [hunk 5]
```diff
@@ -108,6 +108,52 @@ jobs:
...
+
+          # Download and cache dependencies
+          - restore_cache:
+              keys:
+                - v1.3-dependencies-{{ checksum "requirements/combined.txt" }}-{{ checksum "package.json" }}
+                # fallback to using the latest cache if no exact match is found
+                l- v1.3-dependencies-
+
+          - run:
+              command: ./scripts/test_db.sh
```
Comment (iamareebjamal) [inline .circleci/config.yml:145]: Not needed in dredd, not needed here

File: .circleci/config.yml [hunk 6]
```diff
@@ -108,6 +108,52 @@ jobs:
...
+
+          - run:
+              command: pytest tests -v --cov=.
+
+          - run:
+              command: bash <(curl -s https://codecov.io/bash)
+              when: on_success
+          - run:
+              command: bash scripts/push_api_docs.sh
```
Comment (iamareebjamal) [inline .circleci/config.yml:154]: Won't work, remove

File: .circleci/config.yml [hunk 7]
```diff
@@ -119,3 +165,12 @@ workflows:
...
           requires:
             - dependencies
+      - test:
+          requires:
+            - dependencies
+          filters:
+            branches:
+                only:
+                  - master
+                  - development
```
Comment (iamareebjamal) [inline .circleci/config.yml:175]: Remove the filters for now

File: .circleci/config.yml (hunk 4 above)
Comment (codedsun) [reply .circleci/config.yml:119]: env was set in travis file, I copied them

**Comment by iamareebjamal** [comment]
https://app.circleci.com/pipelines/github/fossasia/open-event-server/1397/workflows/8463ea6e-acd8-4ac6-9e36-34154623454c/jobs/5498

File: .circleci/config.yml [hunk 8]
```diff
@@ -108,6 +108,44 @@ jobs:
             - ./.pytype
           key: v1-pytype
 
+  test:
+      docker:
+          - image: circleci/python:3.7.4-node
+            environment:
+              APP_CONFIG: config.TestingConfig
+              DATABASE_URL: postgresql://postgres@localhost/test
+              TEST_DATABASE_URL: postgresql://postgres@localhost/test
```
Comment (iamareebjamal) [inline .circleci/config.yml:117]: You're still using wrong indentation and mixing and matching the amount of spaces. You can literally copy and paste the dredd job section and make edits. And please turn off auto indentation of your IDE.

Comment (iamareebjamal) [reply .circleci/config.yml:117]: Use vim if the IDE is not stopping "intelligently" adding random indentation of spaces which it thinks is correct

Comment (codedsun) [reply .circleci/config.yml]: i fixed it

Comment (codedsun) [reply .circleci/config.yml]: IDE was pycharm was auto indenting on commit

**Comment by codedsun** [comment]
@iamareebjamal - Circle Ci pytest shows

```
#!/bin/bash -eo pipefail
pytest tests -v --cov=.
/bin/bash: pytest: command not found
```

**Comment by iamareebjamal** [comment]
https://github.com/fossasia/open-event-server/pull/7463/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R77

**Comment by iamareebjamal** [comment]
venv is not activated. Look at dredd or pytype job

File: .circleci/config.yml [hunk 9]
```diff
@@ -108,6 +108,43 @@ jobs:
...
+
+      # Download and cache dependencies
+      - restore_cache:
+          keys:
+            - v1.3-dependencies-{{ checksum "requirements/combined.txt" }}-{{ checksum "package.json" }}
+            # fallback to using the latest cache if no exact match is found
+            - v1.3-dependencies-
+
+      - run:
+          command: pytest tests -v --cov=.
```
Comment (iamareebjamal) [inline .circleci/config.yml:142]:
```suggestion
      - run:
          name: Test
          command: . venv/bin/activate && pytest tests -v --cov=.
```

File: .circleci/config.yml [hunk 10]
```diff
@@ -108,6 +108,44 @@ jobs:
             - ./.pytype
           key: v1-pytype
 
+  test:
+    docker:
+      - image: circleci/python:3.7.4-node
+        environment:
+          APP_CONFIG: config.TestingConfig
+          DATABASE_URL: postgresql://postgres@localhost/test
+          TEST_DATABASE_URL: postgresql://postgres@localhost/test
```
Comment (iamareebjamal) [inline .circleci/config.yml:117]:
```suggestion
          TEST_DATABASE_URL: postgresql://postgres@localhost/test
          SECRET_KEY: super secret key
```

**Pull request merged**